eth/backend.go

```diff
@@ -459,6 +459,23 @@ func New(stack *node.Node, config *ethconfig.Config, logger log.Logger) (*Ethere
 		backend.stagedSync.UnwindTo(config.BadBlock-1, badHash)
 	}
 
+	emptyBadHash := config.BadHash == common.Hash{}
+	if !emptyBadHash {
+		var badBlock *types.Block
+		if err = chainKv.View(context.Background(), func(tx kv.Tx) error {
+			block, rErr := rawdb.ReadBlockByHash(tx, config.BadHash)
```

Comment:
Here, you don't need to read the block, it is enough to read header, to find out the block number